lib/Auth/Process/PerunIdentity.php

```diff
@@ -145,10 +145,27 @@ public function process(&$request)
 		}
 
 		$groups = $this->adapter->isUserOnFacility($spEntityId,$user->getId());
+		$serviceName = $request['SPMetadata']['name']['en'];
+
+		if ($serviceName != null){
+			$serviceNameString = $serviceName . " ( " . $spEntityId . " )";
+		} else {
+			$serviceNameString = $spEntityId;
+		}
 
 		if (empty($groups)) {
-			SimpleSAML_Logger::warning('Perun user with identity/ies: '. implode(',', $uids) .' is not member of any assigned group for resource (' . $spEntityId . ')');
-						$this->unauthorized($request);
+			$time = time();
+			$g = array();
+			foreach ($spGroups as $spGroup){
+				array_push($g, "id: " . $spGroup->getId() . ", name: " . $spGroup->getName());
+			}
+			SimpleSAML_Logger::warning('[' . $time . '] Perun user with identity/ies: '.
+				implode(',', $uids) .' is not member of any assigned group for resource (' . $spEntityId . ')');
+			$message = '['. $time .'] Perun user with identity/ies: ' . '<br>*'
```

Comment:
Unfortunately we cannot reveal the name of the groups. So please use this wording:

You are not authorised to access the service ' . $serviceNameString . '. If you feel you should have an access contact $supportContact.

Instead of $supportContact try to use technicalcontact_email from config.php